CLAUDE.md

```diff
@@ -0,0 +1,760 @@
+- Values from environment variables override configuration file values
+
+## CLI Usage
+
+### beacond Commands
+```bash
+beacond init                                    # Initialize a new node
+beacond start                                   # Start the beacon node
+beacond rollback                                # Rollback blockchain state
+beacond genesis add-premined-deposit            # Add premined deposits to genesis
+beacond genesis collect-premined-deposits       # Collect premined deposits
+beacond genesis set-deposit-storage             # Set deposit contract storage
+beacond genesis execution-payload               # Generate execution payload
+beacond deposit create-validator                # Create validator deposit
+```
+
+### Key Flags
+```bash
+--beacon-kit.chain-spec <spec>                  # Chain spec: devnet/testnet/mainnet
+--beacon-kit.chain-spec-file <path>             # Custom chain spec TOML file
+--beacon-kit.engine.jwt-secret-path <path>      # JWT secret for EL auth
+--beacon-kit.engine.rpc-dial-url <url>          # Execution client RPC URL
+--beacon-kit.kzg.trusted-setup-path <path>      # KZG trusted setup file
+--beacon-kit.node-api.enabled                   # Enable REST API
+--home <path>                                   # Node home directory (default: ~/.beacond)
+```
+
+## Network Specifications
+
+### Supported Networks
+- **Devnet** - Chain ID: 80087 (local development)
+- **Testnet/Bepolia** - Chain ID: 80069 (public testnet)
+- **Mainnet** - Chain ID: 80094 (production)
+
+Network configurations are in `testing/networks/<chain-id>/`
+
+## Architecture Overview
+
+BeaconKit implements a modular EVM consensus client using a modified CometBFT for consensus and supporting all major Ethereum execution clients through the Engine API.
+
+### System Architecture
+
+```
+┌───────────────────────────────────────────────────────────────┐
+│                      Service Registry                         │
+│              (Lifecycle orchestrator for all services)        │
+│  Startup Order:                                               │
+│  1. ShutdownService → 2. ValidatorService → 3. NodeAPIServer  │
+│  4. ReportingService → 5. TelemetryService → 6. EngineClient  │
+│  7. ChainService → 8. CometBFTService                         │
+└──────────────────────────────┬────────────────────────────────┘
+                               │ manages lifecycle
+                               ▼
+                ┌─────────────────────────────────────┐
+                │         CometBFT Service            │
+                │  (Consensus orchestrator via P2P)   │
+                │  ┌─────────────────────────────┐    │
+                │  │ ABCI++ Interface:           │    │
+                │  │ • InitChain                 │    │
+                │  │ • PrepareProposal           │    │
+                │  │ • ProcessProposal           │    │
+                │  │ • FinalizeBlock             │    │
```

Comment:
here Commit is missing. not sure if relevant.
Another relevant callback we are have Info, used at the beginning to understand the app state.
The rests are either queries or not implemented.